### PR TITLE
Support for all Symfony translation versions

### LINKED ISSUE #321: symfony/translation 4.2
Hi I use Symfony 4.2, thre is many packages example  
```
 "name": "symfony/twig-bridge",
            "version": "v4.2.10",
```

it have conflict 

```
 "conflict": {
        ...
                "symfony/translation": "<4.2"
            },

```
So i cannot install php-translations. 

Is possible to allow symfony/translation 4.2 + ?

Problem 1
    - Installation request for php-translation/symfony-bundle 0.8.2 -> satisfiable by php-translation/symfony-bundle[0.8.2].
    - php-translation/symfony-bundle 0.8.2 requires symfony/translation ^2.7 || ^3.0 || ^4.0,<4.2 -> no matching package found.

my composer

```
 "minimum-stability": "dev",
    "prefer-stable": true,
    "require": {
        "php": "^7.2",
        "ext-ctype": "*",
        "ext-iconv": "*",
        "doctrine/doctrine-migrations-bundle": "^2.0",
        "friendsofsymfony/ckeditor-bundle": "^1.0",
        "friendsofsymfony/comment-bundle": "^2.2",
        "friendsofsymfony/rest-bundle": "^2.5",
        "friendsofsymfony/user-bundle": "^2.1",
        "grekpg/OhGoogleMapFormTypeBundle": "dev-master",
        "helios-ag/fm-elfinder-bundle": "^9.2",
        "ifsnop/mysqldump-php": "^2.7",
        "jms/serializer-bundle": "^2",
        "knplabs/doctrine-behaviors": "^1.5",
        "knplabs/knp-gaufrette-bundle": "^0.5.3",
        "knplabs/knp-menu-bundle": "^2.2",
        "knplabs/knp-paginator-bundle": "^2.8",
        "liip/imagine-bundle": "^2.1",
        "nelmio/api-doc-bundle": "^3.4",
        "nesbot/carbon": "^2.21",
        "omines/datatables-bundle": "^0.3.0",
        "payum/core": "^1.5",
        "payum/offline": "^1.5",
        "payum/paypal-express-checkout-nvp": "^1.5",
        "payum/payum-bundle": "^2.3",
        "php-http/guzzle6-adapter": "^1.1",
        "php-imap/php-imap": "^3.0.16",
        "secit-pl/imap-bundle": "^1.3",
        "sensio/framework-extra-bundle": "^5.2",
        "sg/datatablesbundle": "^1.0",
        "sonata-project/admin-bundle": "^3.48",
        "sonata-project/doctrine-orm-admin-bundle": "^3.9",
        "sonata-project/page-bundle": "^3.11",
        "sonata-project/translation-bundle": "^2.4",
        "sonata-project/user-bundle": "^4.3",
        "stof/doctrine-extensions-bundle": "^1.3",
        "symfony/console": "4.2.*",
        "symfony/dotenv": "4.2.*",
        "symfony/flex": "^1.1",
        "symfony/framework-bundle": "4.2.*",
        "symfony/swiftmailer-bundle": "^3.2",
        "symfony/translation": "^4.0",
        "symfony/webpack-encore-bundle": "^1.6",
        "symfony/webpack-encore-pack": "^1.0",
        "symfony/yaml": "4.2.*",
        "tbbc/money-bundle": "dev-master",
        "white-october/pagerfanta-bundle": "^1.2"
    },
    "require-dev": {
        "symfony/debug-pack": "*",
        "symfony/dotenv": "^4.1",
        "symfony/maker-bundle": "^1.0",
        "symfony/profiler-pack": "*",
        "symfony/test-pack": "^1.0",
        "symfony/web-server-bundle": "^4.1"
    },


```

### PR DESCRIPTION
**Pull request by Nyholm**
This will replace #323 and fix #321

I basically rewrote and simplified ReplaceOperation.

### CODE REVIEW

File: Catalogue/Operation/ReplaceOperation.php [hunk 1]
```diff
@@ -36,40 +37,72 @@ protected function processDomain($domain)
             'new' => [],
             'obsolete' => [],
         ];
-        $sourceMessages = $this->source->all($domain);
+        if (defined(sprintf('%s::INTL_DOMAIN_SUFFIX', MessageCatalogueInterface::class))) {
```
Comment (XWB) [inline Catalogue/Operation/ReplaceOperation.php]: How about using https://www.php.net/manual/en/function.property-exists.php instead of defined?

Comment (Nyholm) [reply Catalogue/Operation/ReplaceOperation.php]: I don't think `property_exists` works for constants.

Comment (XWB) [reply Catalogue/Operation/ReplaceOperation.php]: You are right.

**Comment by Nyholm** [comment]
Thank you for merging.

**Comment by bocharsky-bw** [comment]
Thank you for working on it! 🎉

**Comment by bobdenotter** [comment]
Thank you for this release! Works like a charm in our project! 

Screenshot 2019-09-05 at 20 45 46

**Pull request merged**